### PR TITLE
Prince/ added accordion component

### PR DESCRIPTION
**Pull request by prince-deriv**
## Update:
Added `Accordion` component
Added `Accordion` with tab

## Variants
`Elevate` - has an elevation effect, containing all properties from the base accordion.
`Fill` - background filled, containing all properties from the base accordion.
`Flush` - also known as FAQ accordion, has the `divider` property.
`Outline` - bordered accordion, containing all properties from the base accordion.

### CODE REVIEW

File: src/components/accordion/base/index.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        </div>
+      </div>
+      <div
+        className={qtMerge(
+          'max-h-[0px] overflow-hidden',
+          accordionTransitionClassNames,
+          (isAutoExpand || isExpanded) && 'max-h-[9999px]',
+        )}
+      >
+        <div className="flex h-fit p-general-lg">{Content && <Content />}</div>
```
Comment (likhith-deriv) [inline src/components/accordion/base/index.tsx:143]: Should we keep the `div` if `Content` is not provided?

File: src/components/accordion/types.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+export interface AccordionProps extends AccordionPropsExcludingNull {
+  id?: string
+  className?: string
+  title?: string
+  subtitle?: string
+  icon?: ReactNode
+  expanded?: boolean
+  divider?: AccordionDivider
+  contentClassname?: string
+  content?: () => ReactNode
```
Comment (likhith-deriv) [inline src/components/accordion/types.ts:15]: What is the difference between `content` and `customContent`  ?

File: src/components/accordion/base/index.tsx (hunk 1 above)
Comment (prince-deriv) [reply src/components/accordion/base/index.tsx]: Yes, it needs to have a default padding.

File: src/components/accordion/types.ts (hunk 2 above)
Comment (prince-deriv) [reply src/components/accordion/types.ts:15]: Needs to re-think the name here, designers are calling it `customContent` which means overriding the whole head of the `accordion` to a custom content of consumer's choice.

**Pull request merged**